src/xrc-tests/src/tests/misbehavior.rs

```diff
@@ -0,0 +1,387 @@
+            base_asset: eur_asset.clone(),
+            quote_asset: btc_asset,
+            timestamp: timestamp_seconds,
+            rate: 22985171,
+            metadata: ExchangeRateMetadata {
+                decimals: 9,
+                base_asset_num_queried_sources: 11,
+                base_asset_num_received_rates: 11,
+                quote_asset_num_queried_sources: 7,
+                quote_asset_num_received_rates: 7,
+                standard_deviation: 1692069,
+                forex_timestamp: Some(yesterday_timestamp_seconds),
+            },
+        };
+
+        let fiat_crypto_pair_result = container
+            .call_canister::<_, GetExchangeRateResult>(
+                "get_exchange_rate",
+                &fiat_crypto_pair_request,
+            )
+            .expect("Failed to call canister for rates");
+        let exchange_rate = fiat_crypto_pair_result
+            .expect("Failed to retrieve an exchange rate from the canister.");
+
+        assert_eq!(exchange_rate, expected_fiat_crypto_pair_rate);
+        assert!(FIAT_CRYPTO_PAIR_BASIC_STD_DEV < exchange_rate.metadata.standard_deviation);
+
+        // Fiat Pair
+        let fiat_pair_request = GetExchangeRateRequest {
+            timestamp: Some(timestamp_seconds),
+            base_asset: eur_asset.clone(),
+            quote_asset: Asset {
+                symbol: "JPY".to_string(),
+                class: AssetClass::FiatCurrency,
+            },
+        };
+
+        let expected_fiat_pair_rate = ExchangeRate {
+            base_asset: eur_asset,
+            quote_asset: Asset {
+                symbol: "JPY".to_string(),
+                class: AssetClass::FiatCurrency,
+            },
+            timestamp: yesterday_timestamp_seconds,
+            rate: 143_028_398_099,
+            metadata: ExchangeRateMetadata {
+                decimals: 9,
+                base_asset_num_queried_sources: 11,
+                base_asset_num_received_rates: 11,
+                quote_asset_num_queried_sources: 11,
+                quote_asset_num_received_rates: 11,
+                standard_deviation: 616_370_071,
+                forex_timestamp: Some(yesterday_timestamp_seconds),
+            },
+        };
+
+        let fiat_pair_result = container
+            .call_canister::<_, GetExchangeRateResult>("get_exchange_rate", &fiat_pair_request)
+            .expect("Failed to call canister for rates");
+        let exchange_rate =
+            fiat_pair_result.expect("Failed to retrieve an exchange rate from the canister.");
+        assert_eq!(exchange_rate, expected_fiat_pair_rate);
+        // Compare the rate's standard deviation to the common (non-malicious) data set.
+        // The non-malicious dataset should be shown that it has lower variability than the malicious set
+        // provided above.
+        assert!(FIAT_PAIR_COMMON_DATASET_STD_DEV < exchange_rate.metadata.standard_deviation);
```

Comment:
The standard deviation is not necessarily larger, right? If the wrong rates are completely off, they will be discarded and do not matter for the calculation of the standard deviation. In this case, the standard deviation remains basically unchanged. It is only larger if the rates are off but not enough to be discarded. I think this point needs to be made clearer.